src/api.py

```diff
@@ -246,14 +249,49 @@ def get_default_agent(self, throw_if_missing: bool = True) -> Optional[Agent]:
         return sub_agent
 
 
+class GameREPL(AgentREPL):
+    last_seen_block = 0
+
+    def print_object_or_objects(
+        self, output: Union[List, Any], metadata: Optional[Dict[str, Any]] = None
+    ):
+        context = AgentContext.get_or_create(
+            client=self.agent_instance.client,
+            context_keys={"id": "default"},
+            searchable=False,
+        )
+        for block in context.chat_history.file.blocks:
+            if block.index_in_file > self.last_seen_block:
+                if block.stream_state == StreamState.STARTED:
+                    while block.stream_state not in [
+                        StreamState.COMPLETE,
+                        StreamState.ABORTED,
+                    ]:
+                        time.sleep(0.4)
```

Comment:
question(non-blocking): do you think in the repl that it is "worth it" to decrease the sleep time?